operations/src/test/java/edu/unc/lib/boxc/operations/impl/metadata/ExportMetadataCsvServiceTest.java

```diff
@@ -67,25 +75,78 @@ void closeService() throws Exception {
     }
 
     @Test
-    public void metadataCsvTest() throws Exception {
+    public void exportDominoMetadataTest() throws Exception {
+        var collectionRecord = makeRecord(COLLECTION_UUID, ADMIN_UNIT_UUID, ResourceType.Collection,
+                "Collection", null, null, null);
         var workRecord1 = makeWorkRecord(UUID1, "Work 1");
         var workRecord2 = makeWorkRecord(UUID2, "Work 2");
-        mockResults(workRecord1, workRecord2);
-        when(solrSearchService.getSearchResults(any()))
-                .thenReturn(makeResultResponse(workRecord1))
-                .thenReturn(makeResultResponse(workRecord2));
-
-        var resultPath = csvService.exportCsv(asPidList(UUID1, UUID2), agent);
-        var csvRecords = parseCsv(ExportMetadataCsvService.CSV_HEADERS, resultPath);
-        assertContainsEntry(csvRecords, "ref_id", UUID1, "Work 1");
-        assertContainsEntry(csvRecords, "ref_id", UUID2, "Work 2");
+        mockParentResults(collectionRecord);
+        mockChildrenResults(workRecord1, workRecord2);
+
+        var resultPath = csvService.exportCsv(asPidList(COLLECTION_UUID), agent);
+        var csvRecords = parseCsv(ExportDominoMetadataService.CSV_HEADERS, resultPath);
+        assertContainsEntry(csvRecords, UUID1, "ref_id", "Work 1");
+        assertContainsEntry(csvRecords, UUID2, "ref_id", "Work 2");
         assertNumberOfEntries(2, csvRecords);
     }
+    
+    @Test
+    public void recordNotFoundTest() {
```

Comment:
Is this testing that the parent is not found? It might be good to clarify that in the name.

It shouldn't be expecting an NPE to be thrown, it looks like `NotFoundException` is the expected exception for the parent being null. The NPE is actually being thrown by the mocking setup rather than running exportCsv. We don't actually need to mock anything to have the parent come back as null, but for clarity we could mock `solrSearchService.getObjectById` since that is how the parent is retrieved.

Also, `assertTrue(Files.notExists(csvPath));` will never be reached since `exportCsv` is expected to throw an exception. Since this is using a temp file for the CSV, we can't predict what the file path will be either, so we can remove the assertion.